storage/filesystem/src/main/java/io/aiven/kafka/tieredstorage/storage/filesystem/FileSystemStorage.java

```diff
@@ -59,21 +60,21 @@ public long upload(final InputStream inputStream, final String key) throws Stora
     }
 
     @Override
-    public InputStream fetch(final String key) throws StorageBackendException {
+    public InputStream fetch(final ObjectKey key) throws StorageBackendException {
         try {
-            final Path path = fsRoot.resolve(key);
+            final Path path = fsRoot.resolve(key.value());
             return Files.newInputStream(path);
         } catch (final NoSuchFileException e) {
-            throw new KeyNotFoundException(this, key, e);
+            throw new KeyNotFoundException(this, key);
```

Comment:
This is needed to satisfy the new tests for masking in `BaseStorageTest`. I don't think we that badly need this inner exception in the file system storage implementation to make the test more complex, so I just decided to remove it. The same below.